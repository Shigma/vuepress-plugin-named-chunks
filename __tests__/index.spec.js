/**
 * @jest-environment node
 */

const { resolve } = require('path')
const { readFileSync } = require('fs')
const { createApp } = require('@vuepress/core')

describe('named-chunks', () => {
  const app = createApp({
    sourceDir: resolve(__dirname, 'docs'),
  })

  beforeAll(async () => {
    await app.process()
    await app.build()
  }, 60000)

  // function testForFile (name, file = name) {
  //   test(name, () => {
  //     const html = readFileSync(resolve(app.outDir, file), 'utf8')
  //       .replace(/(src|href)="\/[\w./]+"/g, (_, $1) => $1 + '="/some/asset"')
  //     expect(html).toMatchSnapshot()
  //   })
  // }

  // testForFile('404.html')
  // testForFile('index.html')
  // testForFile('noscript.html')
  test('hello', () => {
    expect(1).toBe(1)
  })
})
