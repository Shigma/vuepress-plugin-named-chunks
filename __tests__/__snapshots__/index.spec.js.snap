// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`named-chunks 404.html 1`] = `

<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport"
          content="width=device-width,initial-scale=1"
    >
    <title>
      VuePress
    </title>
    <meta name="description"
          content
    >
    <link rel="preload"
          href="/assets/css/0.styles.css"
          as="style"
    >
    <link rel="preload"
          href="/assets/js/styles.js"
          as="script"
    >
    <link rel="preload"
          href="/assets/js/app.js"
          as="script"
    >
    <link rel="preload"
          href="/assets/js/layout-NotFound.js"
          as="script"
    >
    <link rel="prefetch"
          href="/assets/js/7.js"
    >
    <link rel="prefetch"
          href="/assets/js/layout-Custom.js"
    >
    <link rel="prefetch"
          href="/assets/js/layout-Layout.js"
    >
    <link rel="prefetch"
          href="/assets/js/page-8daa1a0e.js"
    >
    <link rel="prefetch"
          href="/assets/js/page-Custom.js"
    >
    <link rel="stylesheet"
          href="/assets/css/0.styles.css"
    >
  </head>
  <body>
    <div id="app">
    </div>
    <script src="/assets/js/styles.js"
            defer
    >
    </script>
    <script src="/assets/js/layout-NotFound.js"
            defer
    >
    </script>
    <script src="/assets/js/app.js"
            defer
    >
    </script>
  </body>
</html>

`;

exports[`named-chunks custom.html 1`] = `

<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport"
          content="width=device-width,initial-scale=1"
    >
    <title>
      VuePress
    </title>
    <meta name="description"
          content
    >
    <link rel="preload"
          href="/assets/css/0.styles.css"
          as="style"
    >
    <link rel="preload"
          href="/assets/js/styles.js"
          as="script"
    >
    <link rel="preload"
          href="/assets/js/app.js"
          as="script"
    >
    <link rel="preload"
          href="/assets/js/layout-Custom.js"
          as="script"
    >
    <link rel="preload"
          href="/assets/js/page-Custom.js"
          as="script"
    >
    <link rel="prefetch"
          href="/assets/js/7.js"
    >
    <link rel="prefetch"
          href="/assets/js/layout-Layout.js"
    >
    <link rel="prefetch"
          href="/assets/js/layout-NotFound.js"
    >
    <link rel="prefetch"
          href="/assets/js/page-8daa1a0e.js"
    >
    <link rel="stylesheet"
          href="/assets/css/0.styles.css"
    >
  </head>
  <body>
    <div id="app"
         data-server-rendered="true"
    >
      <div class="content default">
        <p>
          Custom
        </p>
      </div>
      <div class="global-ui">
      </div>
    </div>
    <script src="/assets/js/styles.js"
            defer
    >
    </script>
    <script src="/assets/js/layout-Custom.js"
            defer
    >
    </script>
    <script src="/assets/js/page-Custom.js"
            defer
    >
    </script>
    <script src="/assets/js/app.js"
            defer
    >
    </script>
  </body>
</html>

`;

exports[`named-chunks index.html 1`] = `

<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport"
          content="width=device-width,initial-scale=1"
    >
    <title>
      VuePress
    </title>
    <meta name="description"
          content
    >
    <link rel="preload"
          href="/assets/css/0.styles.css"
          as="style"
    >
    <link rel="preload"
          href="/assets/js/styles.js"
          as="script"
    >
    <link rel="preload"
          href="/assets/js/app.js"
          as="script"
    >
    <link rel="preload"
          href="/assets/js/layout-Layout.js"
          as="script"
    >
    <link rel="preload"
          href="/assets/js/page-8daa1a0e.js"
          as="script"
    >
    <link rel="prefetch"
          href="/assets/js/7.js"
    >
    <link rel="prefetch"
          href="/assets/js/layout-Custom.js"
    >
    <link rel="prefetch"
          href="/assets/js/layout-NotFound.js"
    >
    <link rel="prefetch"
          href="/assets/js/page-Custom.js"
    >
    <link rel="stylesheet"
          href="/assets/css/0.styles.css"
    >
  </head>
  <body>
    <div id="app"
         data-server-rendered="true"
    >
      <div class="content default">
        <p>
          readme
        </p>
      </div>
      <div class="global-ui">
      </div>
    </div>
    <script src="/assets/js/styles.js"
            defer
    >
    </script>
    <script src="/assets/js/layout-Layout.js"
            defer
    >
    </script>
    <script src="/assets/js/page-8daa1a0e.js"
            defer
    >
    </script>
    <script src="/assets/js/app.js"
            defer
    >
    </script>
  </body>
</html>

`;
